hoomd/md/AlchemyData.h

```diff
@@ -0,0 +1,191 @@
+        }
+
+    void setMass(Scalar new_mass)
+        {
+        mass.x = new_mass;
+        mass.y = Scalar(1.) / new_mass;
+        }
+
+    Scalar getMass()
+        {
+        return mass.x;
+        }
+
+    Scalar getValue()
+        {
+        return value;
+        }
+
+    pybind11::array_t<Scalar> getDAlphas()
+        {
+        ArrayHandle<Scalar> h_forces(m_alchemical_derivatives,
+                                     access_location::host,
+                                     access_mode::read);
+        return pybind11::array(m_alchemical_derivatives.getNumElements(), h_forces.data);
+        }
+
+    Scalar momentum = 0.; // the momentum of the particle
+    Scalar2 mass
+        = make_scalar2(1.0,
+                       1.0); // mass (x) and it's inverse (y) (don't have to recompute constantly)
+    Scalar mu = 0.;          //!< the alchemical potential of the particle
+    GlobalArray<Scalar> m_alchemical_derivatives; //!< Per particle alchemical forces
+    protected:
+    // the timestep the net force was computed and the netforce
+    std::pair<uint64_t, Scalar> m_timestep_net_force;
+    };
+
+struct AlchemicalPairParticle : AlchemicalMDParticle
+    {
+    AlchemicalPairParticle(std::shared_ptr<const ExecutionConfiguration> exec_conf,
+                           int3 type_pair_param)
+        : AlchemicalMDParticle(exec_conf), m_type_pair_param(type_pair_param) {};
+    int3 m_type_pair_param;
+    };
+
+struct AlchemicalNormalizedPairParticle : AlchemicalPairParticle
+    {
+    using AlchemicalPairParticle::AlchemicalPairParticle;
+
+    Scalar alchemical_derivative_normalization_value = 0.0;
+
+    void NormalizeNetForce(Scalar norm_value, Scalar energy_value)
+        {
+        m_timestep_net_force.second *= norm_value;
+        energy_value /= Scalar(m_alchemical_derivatives.getNumElements());
+        m_timestep_net_force.second += energy_value * alchemical_derivative_normalization_value;
+        }
+    };
+
+namespace detail
+    {
+
+inline void export_AlchemicalMDParticles(pybind11::module& m)
+    {
+    pybind11::class_<AlchemicalMDParticle, std::shared_ptr<AlchemicalMDParticle>>(
+        m,
+        "AlchemicalMDParticle")
+        .def_property("mass", &AlchemicalMDParticle::getMass, &AlchemicalMDParticle::setMass)
+        .def_readwrite("mu", &AlchemicalMDParticle::mu)
+        .def_readwrite("alpha", &AlchemicalMDParticle::value)
+        .def_readwrite("momentum", &AlchemicalMDParticle::momentum)
+        .def_property_readonly("forces", &AlchemicalMDParticle::getDAlphas)
+        .def_property_readonly("net_force",
+                               pybind11::overload_cast<>(&AlchemicalMDParticle::getNetForce))
+        .def("notifyDetach", &AlchemicalMDParticle::notifyDetach);
+
+    pybind11::class_<AlchemicalPairParticle,
+                     AlchemicalMDParticle,
+                     std::shared_ptr<AlchemicalPairParticle>>
+        nameThatShouldNeverAppear(m, "AlchemicalPairParticle");
```

Comment:
Me neither... but pretty sure it's included here to avoid a pybind11 error. Probably because I implemented the particles in a likely overly complicated way ... at least at the moment. I did it with the idea that they can be used in the future without a huge rewrite to enable many things that have been talked about but not implemented here.

Reply:
It is not needed.